scripts.py

```diff
@@ -225,89 +230,369 @@ def run(self, data_source_name, metadata_records, epub_directory, cover_director
             work.set_presentation_ready()
             self._db.commit()
 
-class CSVExportScript(Script):
+
+class StaticFeedScript(Script):
+
+    # Identifies csv headers that are not considered titles for lanes.
+    NONLANE_HEADERS = ['urn', 'title', 'author', 'epub']
+
+    @classmethod
+    def header_to_path(cls, header):
+        return [name.strip() for name in header.split('>')]
+
+    @classmethod
+    def category_paths(cls, csv_reader):
+        return filter(
+            lambda h: h.lower() not in cls.NONLANE_HEADERS,
+            csv_reader.fieldnames
+        )
+
+
+class StaticFeedCSVExportScript(StaticFeedScript):
 
     """Exports the primary identifier, title, author, and download URL
     for all of the works from a particular DataSource or DataSources
     to a CSV file format.
+
+    If requested categories are passed in via an existing CSV, this
+    script will add headers to represent those categories, in a proper
+    format for StaticFeedGenerationScript.
     """
 
+    class CategoryNode(object):
```

Comment:
I needed an easier way to access hierarchical information about the proposed lane tree, so I made a class for it. Not sure if this should be inside the StaticFeedCSVExportScript or elsewhere, but it's not being used outside of this class at the moment.